assets/src/stories-editor/blocks/amp-story-page-attachment/attachment-content.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { PostSelector } from '../../components';
+import useElementClickDetector from './element-click-detector';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { RichText } from '@wordpress/block-editor';
+import { Button } from '@wordpress/components';
+import { RawHTML, useEffect, useRef, useState } from '@wordpress/element';
+import { ENTER, SPACE } from '@wordpress/keycodes';
+
+const AttachmentContent = ( props ) => {
+	const [ selectedPost, setSelectedPost ] = useState( null );
+	const [ failedToFetch, setFailedToFetch ] = useState( false );
+	const [ searchValue, setSearchValue ] = useState( '' );
+	let fetchRequest, isStillMounted;
+
+	const {
+		attributes,
+		setAttributes,
+		toggleAttachment,
+	} = props;
+
+	const {
+		attachmentClass,
+		postId,
+		postType,
+		title,
+		wrapperStyle,
+	} = attributes;
+
+	const fetchSelectedPost = () => {
+		isStillMounted = true;
+		if ( postId ) {
+			const currentFetchRequest = fetchRequest = apiFetch( {
+				path: `/wp/v2/${ postType }s/${ postId }`,
+			} ).then(
+				( post ) => {
+					if ( isStillMounted && fetchRequest === currentFetchRequest ) {
+						setSelectedPost( post );
+						setFailedToFetch( false );
+					}
+				}
+			).catch(
+				() => {
+					if ( isStillMounted && fetchRequest === currentFetchRequest ) {
+						setSelectedPost( null );
+						setFailedToFetch( true );
+					}
+				}
+			);
+		}
+	};
+
+	useEffect( () => {
+		fetchSelectedPost();
+		return () => {
+			isStillMounted = false;
+		};
+	}, [] );
+
+	useEffect( () => {
+		fetchSelectedPost();
```

Comment:
Wouldn't this (combined with L66) call this fetch function twice on initial component render? Why do you need both?

Reply:
Not intentional, thank you!